### PR TITLE
Only process executable path for path mapping

### PR DESCRIPTION
**Pull request by fmeum**
This fixes a few issues introduced in 282ac623df3523e2e31a2de9c002e5c50da19fec:
* Only the executable path in the first argument of a `CommandLines` instance should be subject to path mapping.
* String arguments that are not normalized as path strings and thus can't possibly be the optimized memory representation used for the executable arg by `StarlarkAction` should not be modified.

Also removes an class that became unused in that commit.

### CODE REVIEW

**Comment by fmeum** [comment]
@bazel-io flag

**Comment by fmeum** [comment]
(for 8.1.0)

**Review by justinhorvitz — APPROVED** [review]

File: src/test/java/com/google/devtools/build/lib/actions/CommandLinesTest.java [hunk 1]
```diff
@@ -206,4 +209,41 @@ public void expand_flagsOnly_movesOnlyDashDashPrefixedFlagsToParamFile() throws
...
+      builder.addSingleArgument(nonExecutableArg);
+    }
+    CommandLines commandLines = builder.build();
+    PathMapper pathMapper =
+        execPath ->
+            execPath.startsWith(PathFragment.create("bazel-out"))
+                ? PathFragment.create("mapped").getRelative(execPath)
+                : execPath;
+
+    ExpandedCommandLines expanded =
```
[the remark this answers is not in the document]
Comment (fmeum) [reply src/test/java/com/google/devtools/build/lib/actions/CommandLinesTest.java]: Yes

**Comment by fmeum** [comment]
@meteorcloudy If this makes it in time for the next 8.0.0 RC, it would be safer to include it.

**Comment by fmeum** [comment]
@bazel-io fork 8.0.0

**Pull request closed without merging**